boltkit/controller.py

```diff
@@ -287,14 +291,18 @@ def configure(self, properties):
             lines = f_in.readlines()
         with open(config_file_path, "w") as f_out:
             for line in lines:
-                for key, value in properties.items():
+                for key in properties.keys():
                     if line.startswith(key + "=") or \
                             (line.startswith("#") and line[1:].lstrip().startswith(key + "=")):
-                        f_out.write("%s=%s\n" % (key, value))
+                        f_out.write("%s=%s\n" % (key, properties[key]))
+                        del properties[key]
```

Comment:
Modifying a collection while iterating through it can be problematic. Why do you need to remove items?